opencog/atoms/core/Variables.h

```diff
@@ -38,6 +39,20 @@ namespace opencog
  *  @{
  */
 
+// Struct to build paths between variables and root. A path is a
```

Comment:
How is this different from `PatternTerm`?  That device was invented to record the unique path from some root to some atom under that root.

Reply:
I mean this is obviously much more lightweight than that; but I dislike having two mechanisms that are similar-but-different.

Reply:
Yes, it's very lightweight and abstracts away unordered outgoing indices.

I don't like redundancy either, but I think it's justified for now. I've added a note in that regard.